src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/client/util/HomeOfficeDateFormatter.java

```diff
@@ -20,18 +20,43 @@ public static String getCurrentDateTime() {
             DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'"));
     }
 
-    public static String getIacDecisionDate(String hoDecisionDate) {
+    public static String getIacDateTime(String homeOfficeDate) {
         try {
-            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
-            LocalDate parsedDate = LocalDate.parse(hoDecisionDate, formatter);
-            formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
-            return parsedDate.format(formatter);
+            if (homeOfficeDate != null) {
+                //only Date is provided
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                //Date-time is provided
+                if (homeOfficeDate.length() > 10) {
+                    formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
+                }
+                LocalDate parsedDate = LocalDate.parse(homeOfficeDate, formatter);
+                formatter = DateTimeFormatter.ofPattern("dd' 'MMM' 'yyyy");
+                return parsedDate.format(formatter);
+            }
+
         } catch (Exception e) {
             //We return the date we received from Home Office
-            log.info("HO Decision date format error. HO date {}", hoDecisionDate);
+            log.info("HO Decision date format error. HO date {}", homeOfficeDate);
         }
 
-        return hoDecisionDate;
+        return homeOfficeDate;
+    }
+
+    public static String getPersonDateOfBirth(int dayOfBirth, int monthOfBirth, int yearOfBirth) {
+        String hoDateOfBirth = String.format("%02d", dayOfBirth)
+            + "/" + String.format("%02d", monthOfBirth)
+            + "/" + yearOfBirth;
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
```

Comment:
Same as the above. This can be at class level.

Reply:
Changed as suggested